packages/storage/src/fileContentLoader.ts

```diff
@@ -106,26 +109,30 @@ export class FileContentLoader {
     throw error;
   }
 
-  private async loadContent(): Promise<Buffer> {
+  private async loadContent(): Promise<Uint8Array> {
     try {
-      if (Buffer.isBuffer(this.content)) {
+      if (this.content instanceof Uint8Array) {
         return this.content;
       } else if (ArrayBuffer.isView(this.content)) {
-        return Buffer.from(this.content.buffer, this.content.byteOffset, this.content.byteLength);
+        return new Uint8Array(
```

Comment:
I think this copies the underlying ArrayBuffer, not a that big of a deal, but with a large file it might cause issues. 

The constructor `new Uint8Array(buffer, byteOffset, length)` will create a view over the existing data without copying.

Something like:
```ts
return new Uint8Array(this.content.buffer, this.content.byteOffset, this.content.byteLength);'`

Reply:
As long as the input to the Uint8Array constructor is a buffer (e.g. ArrayBuffer here).
But I'm also slicing(/copying) the original buffer (to avoid unwanted side-effects originally, although in this case it doesn't make too much sense (for performance reasons, and also the first if case also returns the original Uint8Array).

Will fix! 👍

[MDN Uint8Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array/Uint8Array#syntax):
`new Uint8Array(buffer, byteOffset, length)`
When called with a `buffer`, and optionally a `byteOffset` and a `length` argument, a new typed array view is created that views the specified ArrayBuffer. The `byteOffset` and `length` parameters specify the memory range that will be exposed by the typed array view. If both are omitted, all of `buffer` is viewed; if only `length` is omitted, the remainder of buffer is viewed.